packages/qwik/src/testing/document.ts

```diff
@@ -20,53 +20,54 @@ export function createDocument(opts?: MockDocumentOptions) {
  * @public
  */
 export function createWindow(opts: MockDocumentOptions = {}): MockWindow {
-  const win = createDocument(opts).defaultView!;
-  return win;
+  return createDocument(opts).defaultView!;
 }
 
 export function ensureGlobals(doc: any, opts?: MockDocumentOptions) {
-  if (!doc[QWIK_DOC]) {
-    if (!doc || doc.nodeType !== 9) {
-      throw new Error(`Invalid document`);
-    }
+  if (!doc || doc.nodeType !== 9) {
+    throw new Error(`Invalid document`);
+  }
+
+  if (doc && doc[QWIK_DOC]) {
+    return doc.defaultView;
```

Comment:
@gioboa  Here returns doc.defaultView

Reply:
I see but we are throwing an error few line before even if doc[QWIK_DOC] is defined. In the old behaviour we were throwing only if not doc[QWIK_DOC]

Reply:
> I see but we are throwing an error few line before even if doc[QWIK_DOC] is defined. In the old behaviour we were throwing only if not doc[QWIK_DOC]

yep. You're right. Updated.